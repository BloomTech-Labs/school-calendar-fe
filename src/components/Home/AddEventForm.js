import React from 'react';
import { useForm } from 'react-hook-form';

// INSERT INTO CREATE_EVENT_MODAL

export default function AddEventForm() {
  const { register, handleSubmit, errors } = useForm();
  const onSubmit = data => console.log(data);
  console.log(errors);
  
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
    <input type="date" placeholder="Start Date" name="Start Date" ref={register({required: true, maxLength: 80})} />
    <input type="time" placeholder="Start Time" name="Start Time" ref={register({required: true, maxLength: 80})} />
    <input type="date" placeholder="End Date" name="End Date" ref={register({required: true, maxLength: 80})} />
    <input type="time" placeholder="End Time" name="End Time" ref={register({required: true, maxLength: 80})} />
    <input type="text" placeholder="Summary" name="Summary" ref={register} />
    <textarea name="Description" ref={register} />

      <input type="submit" />
    </form>
  );
}