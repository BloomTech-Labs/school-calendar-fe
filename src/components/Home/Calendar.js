import React, {useState} from 'react'
import dateFns from 'date-fns'
import { findRenderedDOMComponentWithClass } from 'react-dom/test-utils'

function Calendar() {
    const [ cal , setCal] = useState({
        currentMonth: new Date(),
        selectedDate: new Date ()
    })

    renderHeader() {
        const dateFormat = 'MMMM YYYY';

        return(
            <div className="header row flex-middle">
      <div className="col col-start">
        <div className="icon" onClick={this.prevMonth}>
          chevron_left
        </div>
      </div>
      <div className="col col-center">
        <span>
          {dateFns.format(this.state.currentMonth, dateFormat)}
        </span>
      </div>
      <div className="col col-end" onClick={this.nextMonth}>
        <div className="icon">chevron_right</div>
      </div>
    </div>
        );
    }

    renderDays() {}

    renderCells() {}

    onDateClick = day => {}

    nextMonth = () => {}

    prevMonth = () => {

    }
    return (
        <div className="calendar">
            {renderHeader()}
            {renderDays()}
            {renderCells()}
        </div>
    )
}

export default Calendar
