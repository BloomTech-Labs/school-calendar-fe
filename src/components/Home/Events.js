import React, { useState, useEffect } from "react";
import axios from "axios";


const Events = () => {
  const [data, setData] = useState(null);
  useEffect(() => {
    (async () => {
      const res = await axios.get("/api/events");
      const results = await res.data;
      localStorage.setItem("googleId:", res.data.googleId)
      console.log("results: ", results);
      setData(results);
    })();
  }, [setData]);
  return (
    <div>
      Events
      {/* <div>
        <img src={data && data.photoUrl} alt="" width="90" />
        <div>
          <h6>{data && data.name}</h6>
          <span>{data && data.email}</span>
          <br />
          <span>{data && data.googleId}</span>
        </div>
        <hr />
      </div> */}
      <div>
        
      </div>
      
      <div>
        <ul>
          {data &&
            data.events.map(event => (
              <li key={event.id}>
                <p>
                  {event.start.dateTime}
                  <strong> - {event.summary}</strong>
                </p>
              </li>
            ))}
        </ul>
      </div>
    </div>
  );
};

export default Events;