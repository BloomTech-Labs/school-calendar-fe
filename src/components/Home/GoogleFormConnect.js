import React, {useState} from 'react'
import { gapi } from 'gapi-script'
import { web } from '../../credentials.json'
import { useForm } from 'react-hook-form';


const {API_KEY, client_id} = web


  

const authenticate = () => {
  return gapi.auth2.getAuthInstance()
    .signIn({scope: "https://www.googleapis.com/auth/calendar https://www.googleapis.com/auth/calendar.events"})
    .then(function() { console.log("Sign-in successful"); },
      function(err) { console.error("Error signing in", err); });
}

function loadClient() {
  gapi.client.setApiKey(API_KEY);
  return gapi.client.load("https://content.googleapis.com/discovery/v1/apis/calendar/v3/rest")
    .then(function() { console.log("GAPI client loaded for API"); },
      function(err) { console.error("Error loading GAPI client for API", err); });
}



gapi.load("client:auth2", function() {
  gapi.auth2.init({client_id: client_id});
});

const Blank = () => {
  const [resource, setResource] = useState({
    "start": {
      "dateTime": "" 
    },
    "end": {
      "dateTime": ""
    },
    "description": "",
    "summary": ""
  })

  const dateTime = (date, time) => {
    const DT =  date + 'T' + time + ':00-08:00'+ ''
    console.log(DT)
    return DT
  }
  const { register, handleSubmit, errors } = useForm();
  const onSubmit = data => {
    console.log(data)
    const start =  dateTime(data.startDate, data.startTime) 
    const end = dateTime(data.endDate, data.endTime) 
    setResource({
      "start": {
        "dateTime": start
      },
      "end": {
        "dateTime": end
      },
      "description": data.description,
      "summary": data.summary
    })
    console.log(resource)
  }

  console.log(errors);

  // Make sure the client is loaded and sign-in is complete before calling this method.
function execute(resource) {
  console.log(resource)
  return gapi.client.calendar.events.insert({
    "calendarId": "funnyusernamego@gmail.com",
    "resource": resource
    // "resource": {
    //   "start": {
    //     "dateTime": "2020-01-30T15:00:00-08:00" 
    //   },
    //   "end": {
    //     "dateTime": "2020-01-30T16:00:00-08:00"
    //   },
    //   "description": "Doing Shit",
    //   "summary": "bestlife"
    // }
  })
    .then(function(response) {
      // Handle the results here (response.result has the parsed body).
      console.log("Response", response);
    },
    function(err) { console.error("Execute error", err); });
}


    
  return (
    <div>
      <form onSubmit={handleSubmit(onSubmit)}>
        <input type="date" placeholder="Start Date" name="startDate" ref={register({required: true, maxLength: 80})} />
        <input type="time" placeholder="Start Time" name="startTime" ref={register({required: true, maxLength: 80})} />
        <input type="date" placeholder="End Date" name="endDate" ref={register({required: true, maxLength: 80})} />
        <input type="time" placeholder="End Time" name="endTime" ref={register({required: true, maxLength: 80})} />
        <input type="text" placeholder="Summary" name="summary" ref={register} />
        <textarea name="description" ref={register} />

        <input type="submit" />
      </form>





      <button onClick={() => authenticate().then(loadClient)}>authorize and load</button>
      <button onClick={() => execute(resource)}>execute</button>
    </div>
  )
}
export default Blank