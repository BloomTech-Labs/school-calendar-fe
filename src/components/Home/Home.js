import React from 'react';
import Sidebar from './Sidebar'
import Blank from './Blank'

const Home = () => {

  const style = {
    border: 0,
    width:"900px",
    height:"75vh",
    frameborder:"0",
    scrolling:"no"
  }

  return (
    <div>
      <h1>Home</h1>
      <Sidebar />
      <Blank/>
      <iframe src="https://calendar.google.com/calendar/embed?src=danstad2012%40gmail.com&ctz=America%2FLos_Angeles" style={style}></iframe>
    </div>
  )
}

export default Home;