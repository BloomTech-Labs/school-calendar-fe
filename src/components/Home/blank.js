import React from 'react'
import { gapi } from 'gapi-script'


const authenticate = () => {
  console.log('authenticate!')

  return gapi.auth2.getAuthInstance()
    .signIn({scope: "https://www.googleapis.com/auth/calendar https://www.googleapis.com/auth/calendar.events"})
    .then(function() { console.log("Sign-in successful"); },
      function(err) { console.error("Error signing in", err); });
}

function loadClient() {
  gapi.client.setApiKey("YOUR_API_KEY");
  return gapi.client.load("https://content.googleapis.com/discovery/v1/apis/calendar/v3/rest")
    .then(function() { console.log("GAPI client loaded for API"); },
      function(err) { console.error("Error loading GAPI client for API", err); });
}

function execute() {
  console.log('execute!')
  return gapi.client.calendar.events.insert({
    "calendarId": "danstad2012@gmail.com",
    "resource": {
      "start": {
        "dateTime": "2020-01-29T15:00:00-07:00"
      },
      "end": {
        "dateTime": "2020-01-29T16:00:00-07:00"
      },
      "description": "Doing Shit",
      "summary": "bestlife"
    }
  })
    .then(function(response) {
      // Handle the results here (response.result has the parsed body).
      console.log("Response", response);
    },
    function(err) { console.error("Execute error", err); });
}

const Blank = () => {

  // Make sure the client is loaded and sign-in is complete before calling this method.

  gapi.load("client:auth2", function() {
    gapi.auth2.init({client_id: "YOUR_CLIENT_ID"});
  });

    
  return (
    <div>
      <button onClick={() => authenticate().then(loadClient)}>authorize and load</button>
      <button onClick={() => execute()}>execute</button>
    </div>
  )
}
export default Blank