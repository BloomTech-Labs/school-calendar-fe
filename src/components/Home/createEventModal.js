import React from 'react';

import axios from 'axios';

import {
	makeStyles,
	Modal,
	TextField,
	Typography,
	Button
} from '@material-ui/core';

// modal logic
function rand() {
	return Math.round(Math.random() * 20) - 10;
}

function getModalStyle() {
	const top = 50 + rand();
	const left = 50 + rand();

	return {
		top: `${top}%`,
		left: `${left}%`,
		transform: `translate(-${top}%, -${left}%)`
	};
}

// material-ui styles
const useStyles = makeStyles(theme => ({
	paper: {
		position: 'absolute',
		display: 'flex',
		flexDirection: 'column',
		justifyContent: 'center',
		alignItems: 'center',
		width: 200,
		backgroundColor: theme.palette.background.paper,
		border: '10px solid #cbaa0d',
		boxShadow: theme.shadows[5],
		padding: theme.spacing(2, 4, 3)
	},
	button: {
		background: '#cbaa0d'
	},
	mButton: {
		background: '#2a303d',
		fontSize: '1.3rem',
		padding: 10,
		border: '1px solid #caad0fd4',
		color: '#caad0fd4',
		borderRadius: 10
	}
}));
function createEventModal() {
	const classes = useStyles();

	const [event, setEvent] = useState({
		title: ''
	});

	// getModalStyle is not a pure function, we roll the style only on the first render
	const [modalStyle] = useState(getModalStyle);
	const [open, setOpen] = useState(false);

	const handleChanges = e => {
		setEvent({ ...event, [e.target.name]: e.target.value });
	};

	const handleOpen = () => {
		setOpen(true);
	};

	const handleClose = () => {
		setOpen(false);
	};
	return (
		<div>
			<button type='button' onClick={handleOpen} className={classes.mButton}>
				Create Event
			</button>
			{/* Modal drops down register form */}
			<Modal
				aria-labelledby='simple-modal-title'
				aria-describedby='simple-modal-description'
				open={open}
				onClose={handleClose}
			>
				<div style={modalStyle} className={classes.paper}>
					<Typography>Create Event</Typography>
					<form onSubmit={handleSubmit} className={classes.form}>
						<TextField
							id='outlined-basic'
							label='title'
							name='title'
							margin='normal'
							variant='outlined'
							value={user.title}
							onChange={handleChanges}
							className={classes.input}
							required
						/>
						
						<Button type='submit' className={classes.button}>
							submit
						</Button>
					</form>
				</div>
			</Modal>
		</div>
	);
}

export default createEventModal;
