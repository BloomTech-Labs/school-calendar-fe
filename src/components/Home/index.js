import React, { useState, useEffect } from 'react';
import Calendar from './Calendar/Calendar';
import Logo from '../../img/d8picker.png';
import favicon from '../../img/white.png';
import Template from './Template';
import { useForm } from 'react-hook-form';
import axios from 'axios';
import dayjs from 'dayjs';

// import { axiosWithAuth } from '../../utils/axiosWithAuth';
// import { axiosByGid } from '../../utils/axiosByGid';
import { useTemplate } from '../../hooks/useTemplate';

const Home = ({ profile, eventsApi }) => {
  const {
    selected,
    // setSelected,
    templateList,
    getTemplateList
  } = useTemplate();
  const [data, setData] = useState({});
  const [date, setDate] = useState(dayjs());

  const [events, setEvents] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const [templateFormOpen, setTemplateFormOpen] = useState(false);
  const [formOpen, setFormOpen] = useState(false);

  // January = 0

  const { register, handleSubmit, errors } = useForm();

  // Submit for template form
  const onSubmit = formData => {
    const template = {
      ...formData,
      googleId: localStorage.getItem('googleId')
    };
    console.log('template', template);
    axios
      .post(`${process.env.REACT_APP_ENDPOINT_URL}/api/template`, template)
      .then(res => {
        console.log('Template Post', res);
      })
      .catch(err => {
        console.log(err);
      });
    getTemplateList();
  };

  // Load Events from Google Calendar API and set them to state
  const loadEvents = async () => {
    console.log(eventsApi);
    try {
      const results = await eventsApi.listEvents();
      setEvents(results);
      setIsLoading(false);
    } catch (error) {
      console.log(error);
    }
  };
useEffect(() => {
  if(profile){

    console.log('home index', profile)
  localStorage.setItem('googleId', profile.googleId);

  }
},[profile])

  // Invoke function to load Events from Google Calendar API
  // useEffect(() => {
  //   loadEvents();
  //   profile && localStorage.setItem('googleId', profile.googleId);
  //   console.log('home index', profile)
  //   getTemplateList();
  // }, []);

  const applyTemplate = (summary, description, starttime, endtime) => {
    const EventList = selected.map(e => ({
      end: { dateTime: `${e}T${endtime}:00-8:00` },
      start: { dateTime: `${e}T${starttime}:00-8:00` },
      summary: summary,
      description: description
    }));

    console.log(EventList);
    //dateTime: "2020-02-28T08:30:00-08:00"
    //   const dateTime:''
    //
  };

  return (
    <div className="home">
      <div className="navbar">
        <img src={favicon} alt="" className="favicon" />
        <h2>Sign Out</h2>
      </div>
      <main className="main">
        <div className="left">
          <div className="profile">
            <img
              className="profile-img"
              src={profile && profile.photoUrl}
              alt=""
            />
            <h3>{profile && profile.name}</h3>
          </div>
          <div className="template">
            <h2>templates</h2>
            {templateList.map(t => (
              <Template
                key={t.id}
                id={t._id}
                starttime={t.starttime}
                endtime={t.endtime}
                summary={t.summary}
                description={t.description}
                templateFormOpen={templateFormOpen}
                setTemplateFormOpen={setTemplateFormOpen}
                applyTemplate={applyTemplate}
              />
            ))}
            <button onClick={() => setFormOpen(!formOpen)}>
              Create Template
            </button>
            {formOpen && (
              <div className="Form">
                <form onSubmit={handleSubmit(onSubmit)}>
                  <input
                    type="text"
                    placeholder="summary"
                    name="summary"
                    ref={register({ maxLength: 80 })}
                  />
                  <input
                    type="text"
                    placeholder="description"
                    name="description"
                    ref={register({ maxLength: 100 })}
                  />
                  <input
                    type="time"
                    name="starttime"
                    ref={register({ required: true })}
                  />
                  <input
                    type="time"
                    name="endtime"
                    ref={register({ required: true })}
                  />

                  <input type="submit" />
                </form>
              </div>
            )}
          </div>
        </div>
        <div className="right">
          <img src={Logo} alt="logo" className="logo" />
          <Calendar
            events={events}
            templateFormOpen={templateFormOpen}
            setTemplateFormOpen={setTemplateFormOpen}
            date={date}
            setDate={setDate}
          />
        </div>
      </main>
    </div>
  );
};

export default Home;
