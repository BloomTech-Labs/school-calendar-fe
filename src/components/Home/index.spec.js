import React from 'react';
import { shallow } from 'enzyme';
import Home from './index';
import { create, renderer } from 'react-test-renderer';
import axios from 'axios';
import * as rtl from '@testing-library/react';
import ReactDOM from 'react-dom';
import {render} from '@testing-library/react';


jest.mock('axios');


describe('Home', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    rtl.render(Home, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  
  it('should call data', () => { 
  function fetchData(callback){
    setTimeout(() => {
      callback(data);
    }, 500);
  }
});


it('should match snapshot', () => {
  expect(render(<Home/>)).toMatchSnapshot();
});

it('renders the calendar', () => {
  const { getByText } =  render(Home);
   getByText(/Calendar/i);
});






})
