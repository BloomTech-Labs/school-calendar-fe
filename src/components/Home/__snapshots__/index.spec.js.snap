// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should test Home component 1`] = `
<div>
  <header>
    <div
      id="logo"
    >
      <span
        className="icon"
      >
        D8Picker Calendar
      </span>
    </div>
  </header>
  <main>
    <Calendar
      data={Object {}}
      events={Array []}
    />
  </main>
</div>
`;
